### ISSUE TITLE
Uniform tenant/org/user naming

### ISSUE THREAD

**Issue by pracucci** [cortexproject/cortex#2987]
Cortex interchangeably uses the terms "tenant", "org", "user" and their derived forms "tenant ID", "org ID", "user ID" . This generates quite a lot of confusion to newcomers and I would like to propose to standardise it, converging to a single naming.

We could adopt a soft approach to begin with:
1. Uniform the documentation
2. Introduce a policy to always use the decided term for future changes
3. [lower priority] Eventually consider to rename usage in existing logs, metrics, code

Thoughts?

**Comment by bboreham** [comment]
Agreed, I saw someone confused over this yesterday.

"user" is perhaps justifiable but confusing; "org" is better but confusing if one organisation runs multiple systems, e.g. dev and prod.

The API has "X-Scope-OrgID", which might also need changing.

**Comment by pracucci** [comment]
I think **tenant** would be the best option, because:
- It's a multi-tenant system :)
- An organization entity could have multiple tenants
- An user is usually associated with authentication (out of the scope of Cortex) and a tenant could have multiple users

However, in logs and metrics especially we do a large use of `user`: renaming them would be a pain for the final user, and start using "tenant" there along with the existing "user" could introduce even more confusion. I'm puzzled.

**Comment by simon71** [comment]
As a new user to Cortex that's been going though the often steep learning curve and trying to understand the terminology could the confusion not be solved by firstly having some clear documentation about what each means and how each part of the system uses them and then look at writing some of them out of future releases?

**Comment by pracucci** [comment]
> As a new user to Cortex that's been going though the often steep learning curve and trying to understand the terminology could the confusion not be solved by firstly having some clear documentation about what each means and how each part of the system uses them and then look at writing some of them out of future releases?

The main problem is that within Cortex these 3 terms are just synonymous and I believe uniforming it (just pick 1) could help simplifying it.

**Comment by simon71** [comment]
You said 

> _However, in logs and metrics especially we do a large use of user: renaming them would be a pain for the final user, and start using "tenant" there along with the existing "user" could introduce even more confusion._

In the mean time would documenting the current state of play and how thing will change in the future not help?

**Comment by bboreham** [comment]
Some places you will see "user" vs "org" was copy-paste four years ago from code where those two things mean something; however in Cortex today they are never used to mean anything different from "tenant".

**Comment by pracucci** [comment]
> You said
> 
> > _However, in logs and metrics especially we do a large use of user: renaming them would be a pain for the final user, and start using "tenant" there along with the existing "user" could introduce even more confusion._
> 
> In the mean time would documenting the current state of play and how thing will change in the future not help?

Definitely. Uniforming the documentation and mentioning that logs and metrics may use these terms interchangeably is good starting point 👍

**Comment by pracucci** [comment]
A good feedback received from @weeco on Slack:

> Adding a glossary could be an idea as well. It may help new users to describe the meaning of churn, metric series etc. even though they are rather prometheus specific

**Comment by jtlisi** [comment]
+1 for **tenant**

**Comment by jtlisi** [comment]
> However, in logs and metrics especially we do a large use of user: renaming them would be a pain for the final user, and start using "tenant" there along with the existing "user" could introduce even more confusion. I'm puzzled.

I think we should roadmap a hard switch for our next major release. In the meantime we can standardize on `tenant`.

**Comment by roidelapluie** [comment]
still valid

**Comment by pracucci** [comment]
_Still valid, but really low priority._

**Comment by pracucci** [comment]
_Still valid, but very low priority.

**Comment by pracucci** [comment]
_Still valid, but still very very low priority._

**Comment by friedrichg** [comment]
still valid
